dbt_sugar/core/config/config.py

```diff
@@ -112,7 +118,34 @@ def assert_dbt_projects_exist(self) -> bool:
             )
         return True
 
+    def locate_config(self) -> None:
+        folder_iteration = 0
+        logger.debug(f"Starting config file finding from {self._current_folder}")
+        current = self._current_folder
+        filename = Path(current).joinpath(self.SUGAR_CONFIG_FILENAME)
+
+        if self._config_path == Path(str()):
+            logger.debug("Trying to find sygar_config.yml in current and parent folders")
+
+            while folder_iteration < self._max_folder_iterations:
+                if filename.exists():
+                    sugar_config_dir = filename
+                    logger.debug(f"{filename} exists and was retreived.")
+                    self._config_path = sugar_config_dir
+                    self._config_file_found_nearby = True
+                    break
+                current = current.parent
```

Comment:
have you checked if we reach a point which the folder doesn't have a parent?

Reply:
the rest looks really nice

Reply:
Do you mean for example if we're at the root of the hard drive and we cannot go up more?

Reply:
exactly!

Reply:
Right no I haven't haha. That's a good idea though.

Do you have some suggestions as to how to make this better?

Reply:
@virvirlopez I tried to see what happens when you do `Path('/').parent` and you just keep on getting `PosixPath('/')` which is fine. This will run up to 4 times and then we will exit the loop and say that we cannot find the config file from the current directory and up (which in that case is the same).

Do you think we're good on this? I might test it again when we have the config loading hooked up in the flow and we can call `dbt-sugar` from anywhere. But I think it should be no problem at all.